apps/system/lockscreen/js/lockscreen_inputpad.js

```diff
@@ -202,14 +220,18 @@
         }
 
         this.states.passCodeEntered += key;
-        this.updatePassCodeUI();
+        this.updatePassCodeUI('');
 
         if (this.states.padVibrationEnabled) {
           navigator.vibrate(this.configs.padVibrationDuration);
         }
 
         if (this.states.passCodeEntered.length === 4) {
-          this.lockScreen.checkPassCode(this.states.passCodeEntered);
+          (function(value) {
+            setTimeout(function() {
+              this.lockScreen.checkPassCode(value);
+            }, CODE_VALIDATION_TIMEOUT);
```

Comment:
And besides, the user could have pressed backspace and entered a new key in those 1200ms and then you'd be validating the wrong PIN!